src/StaticMock/Mock.cs

```diff
@@ -133,6 +133,165 @@ public static void SetupDefault(Type type, string methodName, Action action)
             MockHelper.SetupDefault(methodToReplace, action);
         }
 
+        public static IFuncMockService Setup(Type type, string methodName, BindingFlags bindingFlags, Action action)
```

Comment:
This and other methods you add is duplicates of already existing methods. You can make method with signature: `private static IFuncMockService Setup(Type type, string methodName, BindingFlags? bindingFlags, Action action)` with nullable BindingFlags check for null and call GetMethod(methodName) or GetMethod(methodName, bindingFlags) and reuse this method in method upper and in this method. If you will have troubles with understanding of my idea. I can write an example.

Reply:
Hi, @SvetlovA , I am not sure your think. please give me a example.
I modified the method, like the following:

<img width="592" alt="キャプチャ" src="https://user-images.githubusercontent.com/48502242/125399015-f956cd80-e3ea-11eb-8fdf-dcac96f575f1.PNG">

Reply:
ok, here is my example:
```
        public static IFuncMockService Setup(Type type, string methodName, Action action) =>
            SetupInternal(type, methodName, action);

        public static IFuncMockService Setup(Type type, string methodName, BindingFlags bindingFlags, Action action) =>
            SetupInternal(type, methodName, action, bindingFlags);

        private static IFuncMockService SetupInternal(Type type, string methodName, Action action, BindingFlags? bindingFlags = null)
        {
            if (type == null)
            {
                throw new ArgumentNullException(nameof(type));
            }

            if (methodName == null)
            {
                throw new ArgumentNullException(nameof(methodName));
            }

            if (action == null)
            {
                throw new ArgumentNullException(nameof(action));
            }

            var methodToReplace = bindingFlags.HasValue
                ? type.GetMethod(methodName, bindingFlags.Value)
                : type.GetMethod(methodName);

            if (methodToReplace == null)
            {
                throw new Exception($"Can't find method {methodName} of type {type.FullName}");
            }

            if (methodToReplace.ReturnType == typeof(void))
            {
                throw new Exception($"Can't use some features of this setup for void return. To Setup void method us {nameof(SetupVoid)} setup");
            }

            return new FuncMockService<object>(new HookServiceFactory(), new HookBuilder(), methodToReplace, action);
        }
```

Reply:
Good morning.  @SvetlovA ， Thank you very much for your example. I don't understand the advantages of your method.
Why can't we just declare functions like this:
 public static IFuncMockService Setup(Type type, string methodName, Action action, BindingFlags? bindingFlags = null)
 {
         ............
 } 
Please tell me your reasons about your example

Reply:
Ok, your case is well, but I have already released previous API version. We mustn't change API, only add new. It something like letter "O" in SOLID. And also I think that "action" must be last parameter. And in case when "action" is last parameter we can't set default value for nullable BindingFlags. Also you can see the example of Microsoft realization of "GetMethod" method in Type (https://docs.microsoft.com/en-us/dotnet/api/system.type.getmethod?view=net-5.0) they use overloads for it. I think it is something like common principles of API creation. But It is only my subjective opinion 😊

Reply:
hi, @SvetlovA .I think you're right. I've modified the code. Please check it. Thank you very much for your guidance.